sdk/core/azure-core/src/main/java/com/azure/core/util/FluxUtil.java

```diff
@@ -427,32 +431,49 @@ public void onSubscribe(Subscription s) {
 
             @Override
             public void onNext(ByteBuffer bytes) {
-                isWriting = true;
-                outFile.write(bytes, pos, null, onWriteCompleted);
+                if (isWriting) {
+                    if (!writeQueue.offer(bytes)) {
+                        onError(new IllegalStateException("Dropped onNext when writing to file."));
+                    }
```

Comment:
Reaching here is a bug in the upstream reactive operator.  This means that there are two or more concurrent `onNext()` calls and `writeQueue.offer()` could potentially be adding `bytes` to the queue out of order. It might be better to just throw an error and stop processing the stream.

Reply:
Remove this, and will update it to emit an error if there is another onNext while writing is in-progress.